assets/src/dashboard/components/cardGridItem/cardPreview.js

```diff
@@ -84,13 +87,75 @@ const getActionAttributes = (targetAction) =>
     ? { href: resolveRoute(targetAction), isLink: true }
     : { onClick: targetAction };
 
-const CardPreviewContainer = ({ centerAction, bottomAction, children }) => {
+const cardMachine = {
+  idle: {
+    activate: 'active',
+  },
+  active: {
+    deactivate: 'idle',
+  },
+};
+
+const cardReducer = (state, action) => cardMachine?.[state]?.[action] || state;
+
+const CardPreviewContainer = ({
+  centerAction,
+  bottomAction,
+  story,
+  children,
+}) => {
   const { pageSize } = usePagePreviewSize({ isGrid: true });
+  const [cardState, dispatch] = useReducer(cardReducer, 'idle');
+  const [pageIndex, setPageIndex] = useState(0);
+  const containElem = useRef(null);
+  const storyPages = story.pages || [];
+
+  useFocusOut(containElem, () => dispatch('deactivate'), []);
+
+  useEffect(() => {
+    if ('idle' === cardState) {
+      setPageIndex(0);
+    }
+  }, [cardState]);
+
+  useEffect(() => {
+    let intervalId;
+    if ('active' === cardState) {
+      /**
+       * The interval duration should eventually get pulled off the story schema's
+       * auto advance duration and if no duration provided, use a default.
+       *
+       * Can also incorporate onWAAPIFinish here to make sure the page
+       * doesn't switch before the animations finishes.
+       */
+      intervalId = setInterval(
+        () => setPageIndex((v) => clamp(v + 1, [0, storyPages.length - 1])),
+        2000
+      );
```

Comment:
Can we make the `2000` a constant somewhere so it's easy to re-configure the min-duration a page should appear before switching without having to dive into the code?